superset-frontend/src/SqlLab/components/SaveQuery/index.tsx

```diff
@@ -98,6 +98,8 @@ const SaveQuery = ({
   const [showSaveDatasetModal, setShowSaveDatasetModal] = useState(false);
   const isSaved = !!query.remoteId;
   const canExploreDatabase = !!database?.allows_virtual_table_explore;
+  const canShowSaveButton =
+    database?.allows_virtual_table_explore !== undefined;
```

Comment:
what should the value of canShowSaveButton be if database is null or database.allows_virtual_table_explore is False?

Reply:
Basically, what's the difference between canExploreDatabase and canShowSaveButton? Also, just a naming nit, I would change canShowSaveButton to something like shouldShowSaveButton, to signify that 1) it's a boolean like you have it, but also 2) that it's not related to permissions but rather state.

Reply:
database should always be set since the parent component <SqlEditor/> returns an <EmptyStateBig/> component if database is not defined.
The allows_virtual_table_explore property is added to the store when dispatch(getDatabases) action is called inside of the <SqlEditorLeftTab/>, and then the <SaveQuery/> component gets rerendered with databases.allows_virtual_table_explore as a boolean.

While the getDatabases action is awaiting a response databases.allows_virtual_table_explore will be undefined, and  canShowSaveButton prevents the save button from being rendered. Once the additional databases data has been added to the redux store, the save button will either render a split button, or the original button based upon the value canExploreDatabase.